Formula/codeclimate.rb

```diff
@@ -1,19 +1,18 @@
-require "formula"
-
 class Codeclimate < Formula
-  CODECLIMATE_VERSION = "0.24.2".freeze
-
+  desc "Code Climate CLI"
   homepage "https://github.com/codeclimate/codeclimate"
-  version CODECLIMATE_VERSION
-
-  url "https://github.com/codeclimate/codeclimate/archive/v#{CODECLIMATE_VERSION}.tar.gz"
-  sha1 "c28ce24aea878ecd8b1bd65cc313206bcd36aece"
+  url "https://github.com/codeclimate/codeclimate/archive/v0.24.2.tar.gz"
+  sha256 "f6ab47d602c8a6ec2d4e9f86df829cc31c3059ad025058c7646499b154242351"
 
   def install
     # Alter PATH to ensure `docker' is available
-    system "env", \
-      "PATH=#{HOMEBREW_PREFIX}/bin:#{ENV["PATH"]}", \
-      "PREFIX=#{prefix}", "make", "install"
+    if Formula["docker"].linked_keg.exist?
```

Comment:
For your first question, it is likely that the user ended up in this situation because they installed docker via homebrew and manually.

For your second question, it is true that the PATH environment variable is isolated during the `brew install` process, see https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Formula-Cookbook.md#check-for-dependencies for details. However, I think the real question is do you want a code climate install via homebrew to require docker via homebrew. I think that if that was the requirement, it would make things a lot cleaner and prevent issues in the future. We could always add `/usr/local/bin` back to the path, but I would shy away from that. There is no reason that we couldn't get the current formula moved to homebrew proper, which would make installation even easier for users. However, adding `/usr/local/bin` to the path would likely prevent the move to homebrew proper from occurring.